File: abc/abc301/b.cpp
// https://atcoder.jp/contests/abc301/submissions/41351553

#include <bits/stdc++.h>
using namespace std;

int main() {
  int n;
  cin >> n;

  vector<int> a(n);
  for (auto &z : a) cin >> z;

  cout << a[0] << " ";

  for (int i = 1; i < a.size(); i++) {
    if (a[i - 1] < a[i]) {
      for (int j = a[i - 1] + 1; j < a[i]; j++) {
        cout << j << " ";
      }
      cout << a[i] << " ";
    } else {
      for(int j = a[i - 1] - 1; j > a[i]; j--) {
        cout << j << " ";
      }
      cout << a[i] << " ";
    }
  }
}

File: abc/abc301/b.py
# https://atcoder.jp/contests/abc301/submissions/41402046

n = int(input())
a = list(map(int, input().split()))

while 1:
  for i in range(len(a) - 1):
    if a[i] - a[i + 1] < -1:
      a = a[:i + 1] + list(range(a[i] + 1, a[i + 1])) + a[i + 1:]
      break

    if a[i] - a[i + 1] > 1:
      a = a[:i + 1] + list(range(a[i] - 1, a[i + 1], -1)) + a[i + 1:]
      break

    if i == len(a) - 2:
      print(*a)
      exit()

